ddtrace/internal/_encoding.pyx

```diff
@@ -46,16 +49,24 @@ cdef inline int PyBytesLike_Check(object o):
     return PyBytes_Check(o) or PyByteArray_Check(o)
 
 
+cdef inline int array_prefix_size(int l):
+    if l < 16:
+        return 1
+    elif l < (2<<16):
+        return 3
+    return MSGPACK_ARRAY_LENGTH_PREFIX_SIZE
```

Comment:
I can't think of a reason not to just use 5 other than to avoid unused space in the buffer, @P403n1x87 do you have any thoughts?

Reply:
If we used 5 every time we would have to manually implement writing the size prefix according to the msgpack specs. If we use the C API for that we need this variable size. I opted for the latter to avoid leaking any msgpack implementation details into our code.